Mage/src/main/java/mage/game/GameState.java

```diff
@@ -819,19 +818,31 @@ public boolean hasSimultaneousEvents() {
     public void addSimultaneousDamage(DamagedEvent damagedEvent, Game game) {
         // combine damages per type (player or permanent)
         boolean flag = false;
+        Set<UUID> flagPerPlayer = new HashSet<>();
         for (GameEvent event : simultaneousEvents) {
             if ((event instanceof DamagedBatchEvent)
                     && ((DamagedBatchEvent) event).getDamageClazz().isInstance(damagedEvent)) {
                 // old batch
                 ((DamagedBatchEvent) event).addEvent(damagedEvent);
                 flag = true;
-                break;
```

Comment:
any side effects from no longer breaking this loop?

Reply:
I do not think so. This was an optimization to find `DamagedBatchEvent` (there can only be a single one in the list at a given time) that can be among the `simultaneousEvent` list. As I'm looking for more batch events in the list, I needed to remove that `break`.